app/src/main/kotlin/com/wire/android/util/keyboard/KeyboardInsetsProvider.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.android.util.keyboard
+
+import android.R
+import android.app.Activity
+import android.content.Context
+import android.graphics.Rect
+import android.view.View
+import android.view.ViewTreeObserver
+import android.view.Window
+import androidx.compose.runtime.compositionLocalOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import kotlin.math.roundToInt
+
+class KeyboardInsetsProvider constructor(
+    applicationContext: Context,
+    activity: Activity
+) : ViewTreeObserver.OnGlobalLayoutListener {
+
+    private val rootWindow: Window = activity.window
+    private val rootDecorView: View = rootWindow.decorView
+    private val rootView: View = rootDecorView.findViewById(R.id.content)
+
+    private val rectWindowVisibleDisplayFrame = Rect()
+
+    private val deviceDensity = applicationContext.resources.displayMetrics.density
+
+    private var screenMaxHeight = 0
+
+    val keyBoardSize: KeyboardSize = KeyboardSize()
+
+    init {
+        rootView.viewTreeObserver.addOnGlobalLayoutListener(this)
```

Comment:
`KeyboardInsertsProvider` holds a reference to `rootView` and `activity.window`, and adds itself as a listener to the `rootView.viewTreeObserver`.

This by itself isn't a leak, but if there's anything holding a reference to this `KeyboardInsetsProvider`, maybe something from `Compose`, this causes a Leak.

I'm not too experienced on how Compose works underneath to understand if `CompositionLocalProvider(LocalKeyboardSize provides keyboardInsetsProvider.keyBoardSize)` will hold `keyboardInsetsProvider` more than it should or not.

Maybe we can play with closing/opening the Keyboard while on the `ConversationScreen` and get some dumps to be sure.

Reply:
actually I did not think about removing this global layout listener, from what I understand 

```
@Composable
@OptIn(InternalComposeApi::class)
fun CompositionLocalProvider(vararg values: ProvidedValue<*>, content: @Composable () -> Unit) {
    currentComposer.startProviders(values)
    content()
    currentComposer.endProviders()
}
```

what it does it kind of places that "provder" into the istance of current composer so that we can access it "globally" and indeed this will i think cause the keyboardInsetsProvider to live as long as the current composer is living, and because the activity will live as long as the current composer will live, not sure what the lifecycle is of the currentComposer, but for safety I'll try to think about a solution that will require to use the keyboardInsertsProvider, also i did not think about removing the listener. 

if you have any more suggetions @vitorhugods  pls let me know